File: bernoulli.erl
%% erl
%% c(bernoulli).
%% bernoulli:main().

-module(bernoulli).

-export([main/0,loop/4, binom/2, bernoulli/0]).

% bernoulli function calculates and prints a list with the 20 first bernoulli numbers
bernoulli() ->
    M = lists:seq(2,20),
    Dict = dict:new(),
    Dict_B = dict:store(1,1,Dict), % accumulating the list B

    Dict_B_complete = lists:foldl(fun(M_counter,Acc_B)->
                            Value = lists:foldl(
                                   fun(K,Bm)-> 
                                       Bm - binom(M_counter, K-1) * dict:fetch(K,Acc_B) end, %B[m] <-- B[m] − BINOM (m, k-1) * B[k]
                                   0, %Bm
                                   lists:seq(1,M_counter-1)), % for k ← 1 to m −1 do (K)
                            dict:store(M_counter,Value/M_counter,Acc_B) %index, value, B
                        end,
                        Dict_B, %Acc_b
                        M % for m <-- 2 to 20 do (M_counter)
                        ),

    io:format("~p",[lists:sort(dict:to_list(Dict_B_complete))]).


% binom calculates the binomial coefficient
binom(N, K) ->
    R = 1,
    I = 1,
    K1 = K + 1,
    R_new = loop(K1, I, R, N),
    R_new.

% Helper function to the binom function
loop(I, I, R, _) ->
    R;
loop(K, I, R, N) ->
    R_new = (R *(N - I + 1))/I,
    loop(K, I+1, R_new, N).

main() -> 
   bernoulli().

File: bernoulli.php
// php bernoulli.php    
// php -r "require 'bernoulli.php'; main();"

<?php
// bernoulli function calculates and prints a list with the 20 first bernoulli numbers
function bernoulli() {
    $list = array();
    $list[0] = 1;
    $n = 20;
    for ($m = 1; $m <= $n; $m++) { 
        $list[$m] = 0;
        for ($k=0; $k <= $m - 1; $k++) { 
            $list[$m] = $list[$m] - binom($m + 1, $k) * $list[$k];
        }
        $list[$m] = $list[$m]/($m + 1);
    }
    return $list;
}
// binom calculates the binomial coefficient
function binom($n, $k) {
    $r = 1;
    for ($i = 1; $i <= $k; $i++) { 
        $r = $r * ($n - $i + 1) / $i;
    }
    return $r;
}

function main(){
    $result = bernoulli();
    foreach($result as $value){
        echo "$value";
        echo "\r\n";
    }
}

?>